lib/rubocop/cop/naming/restricted_variable_name.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Naming
+      # Checks for assigning to a restricted identifier name. Restricted names
+      # are forbidden for local variables, instance variables, class variables,
+      # global variables, method arguments (positional, keyword, rest or block),
+      # and block arguments.
+      #
+      # Method names are not covered by this cop.
+      #
+      # NOTE: The cop is not configured with any restricted identifier names
+      # by default. Assign values to the `Identifiers` array to use the cop.
+      #
+      # @example Identifiers: ['require']
```

Comment:
Probably I'd name this `Names` or `ForbiddenNames`. (identifier is a bit broader term than variable names IMO)